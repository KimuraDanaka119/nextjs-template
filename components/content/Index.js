import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import slug from 'slug';

import withI18N from '../hoc/withI18N';

const Index = ({ t, result }) => (
  <div>
    <div className="index">
      <h1>{t('index:content.header')}</h1>
      <p>{t('index:content.subTitle', { count: result.data.length })}</p>
      <ul>
        {result ?
          result.data.map(item =>
            <li key={item.show.id}>
              <Link prefetch href={`/show?id=${item.show.id}`} as={`/shows/${item.show.id}-${slug(item.show.name)}`}>
                <a>{item.show.name} {item.show.rating.average ? `(${item.show.rating.average})` : ''}</a>
              </Link>
            </li>)
          :
          ''
        }
      </ul>
    </div>
  </div>
);

Index.propTypes = {
  t: PropTypes.func,
  result: PropTypes.object,
};

export default withI18N(Index, ['index', 'error']);
export const undecorated = Index;
