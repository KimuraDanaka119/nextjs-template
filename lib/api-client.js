import fetch from 'isomorphic-unfetch';
import config from 'configs/config';

export default class ApiClient {
  async doFetch(path) {
    const url = config.api.baseURL + path;

    const res = await fetch(url);
    const data = await res.json();

    return {
      statusCode: res.status,
      data,
    };
  }

  async getShows(keyword) {
    const res = await this.doFetch(`/search/shows?q=${keyword}`);

    return res;
  }

  async getShow(id) {
    const res = await this.doFetch(`/shows/${id}`);

    return res;
  }
}
