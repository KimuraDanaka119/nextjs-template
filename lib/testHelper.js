import ApiClient from './api-client';

function t(key) {
  return key;
}

const config = {
  env: 'test',
  versionHash: 'test',
  paths: {
    static: '/static/test',
    styles: '/static/test/styles',
    images: '/static/test/images',
    locales: '/static/test/locales',
  },
  api: {
    baseURL: '.',
  },
};

const apiClient = new ApiClient('.');

export { t, config, apiClient };
