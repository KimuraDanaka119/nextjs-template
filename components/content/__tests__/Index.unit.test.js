import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { undecorated as Index } from '../Index';
import ApiClient from '../../../lib/api-client';
import { t } from '../../../lib/testHelper';

const apiClient = new ApiClient('.');

describe('<Index />', () => {
  it('should render correctly with result', async () => {
    const result = await apiClient.getShows('batman');

    const footer = shallow(<Index t={t} result={result}/>);
    expect(toJson(footer)).toMatchSnapshot();
  });
});
