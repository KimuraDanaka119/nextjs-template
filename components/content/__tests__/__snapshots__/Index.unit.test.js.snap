// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> should render correctly with result 1`] = `
<div>
  <div
    className="index"
  >
    <h1>
      index:content.header
    </h1>
    <p>
      index:content.subTitle
    </p>
    <ul>
      <li
        key="975"
      >
        <Link
          as="/shows/975-Batman"
          href="/show?id=975"
          prefetch={true}
        >
          <a>
            Batman
             
            (8.2)
          </a>
        </Link>
      </li>
      <li
        key="481"
      >
        <Link
          as="/shows/481-The-Batman"
          href="/show?id=481"
          prefetch={true}
        >
          <a>
            The Batman
             
            (8.3)
          </a>
        </Link>
      </li>
      <li
        key="504"
      >
        <Link
          as="/shows/504-Batman-Beyond"
          href="/show?id=504"
          prefetch={true}
        >
          <a>
            Batman Beyond
             
            (9)
          </a>
        </Link>
      </li>
      <li
        key="757"
      >
        <Link
          as="/shows/757-Batman-The-Animated-Series"
          href="/show?id=757"
          prefetch={true}
        >
          <a>
            Batman: The Animated Series
             
            (9.3)
          </a>
        </Link>
      </li>
      <li
        key="11464"
      >
        <Link
          as="/shows/11464-Batman-Unlimited"
          href="/show?id=11464"
          prefetch={true}
        >
          <a>
            Batman Unlimited
             
            (10)
          </a>
        </Link>
      </li>
      <li
        key="3557"
      >
        <Link
          as="/shows/3557-Beware-the-Batman"
          href="/show?id=3557"
          prefetch={true}
        >
          <a>
            Beware the Batman
             
            (10)
          </a>
        </Link>
      </li>
      <li
        key="33618"
      >
        <Link
          as="/shows/33618-The-Adventures-of-Batman"
          href="/show?id=33618"
          prefetch={true}
        >
          <a>
            The Adventures of Batman
             
          </a>
        </Link>
      </li>
      <li
        key="900"
      >
        <Link
          as="/shows/900-Batman-The-Brave-and-The-Bold"
          href="/show?id=900"
          prefetch={true}
        >
          <a>
            Batman: The Brave and The Bold
             
            (9.4)
          </a>
        </Link>
      </li>
      <li
        key="5951"
      >
        <Link
          as="/shows/5951-The-New-Batman-Adventures"
          href="/show?id=5951"
          prefetch={true}
        >
          <a>
            The New Batman Adventures
             
            (8.6)
          </a>
        </Link>
      </li>
      <li
        key="22309"
      >
        <Link
          as="/shows/22309-Batman-Black-and-White"
          href="/show?id=22309"
          prefetch={true}
        >
          <a>
            Batman: Black and White
             
          </a>
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
